numpy/ma/core.py

```diff
@@ -191,14 +191,13 @@ class MaskError(MAError):
     default_filler["M8[" + v + "]"] = np.datetime64("NaT", v)
     default_filler["m8[" + v + "]"] = np.timedelta64("NaT", v)
 
+float_types_list = [np.half, np.single, np.double, np.longdouble,
+                    np.csingle, np.cdouble, np.clongdouble]
 max_filler = ntypes._minvals
-max_filler.update([(k, -np.inf) for k in [np.float32, np.float64]])
+max_filler.update([(k, -np.inf) for k in float_types_list])
```

Comment:
Doesnt look like this bug is still present.
I checked the following

```
>>> import numpy.core.numerictypes as ntypes
>>> np.float16 in ntypes._maxvals
True
>>> np.float16 in ntypes._minvals
>>> np.__version__
'1.19.0.dev0+8f243ca'
>>> test_array = np.ma.array(np.random.rand(10), mask=np.eye(10)[0], dtype=np.float16)
>>> print(test_array)
[-- 0.42431640625 0.5009765625 0.6455078125 0.97119140625 0.2093505859375
 0.83056640625 0.228515625 0.2481689453125 0.91650390625]
>>> print(test_array.max())
0.971
```

Reply:
my bad, i may be looking in the wrong branch! :)

Reply:
When you look at it one more time, you always find something... I guess the `max_filler` should really be `-np.inf- 1j*np.inf`, so that the (strange) sorting and min/max of `np.array([np.inf+3])` (as a masked array with masked values) makes sense.

Reply:
That doesn't work either, `1j*np.inf` gives `nan+infj`

Reply:
Not sure how we can fix this , any further suggestions?

Reply:
eh, I did not realize about that behaviour. But `complex("inf+infj")` works fine (and `cmath.infj` exists, but importing that seems unnecessary. (it seems our complex can only coerce real strings.)